src/main/java/digital/slovensko/autogram/core/visualization/PDFVisualization.java

```diff
@@ -1,32 +1,48 @@
 package digital.slovensko.autogram.core.visualization;
 
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.util.Base64;
+import java.util.ArrayList;
 
 import digital.slovensko.autogram.core.SigningJob;
 import digital.slovensko.autogram.ui.Visualizer;
 import eu.europa.esig.dss.model.DSSDocument;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.rendering.ImageType;
+import org.apache.pdfbox.rendering.PDFRenderer;
+
+import javax.imageio.ImageIO;
 
 public class PDFVisualization extends Visualization {
     private final DSSDocument document;
+    private final int pdfDpi;
+
 
-    public PDFVisualization(DSSDocument document, SigningJob job) {
+    public PDFVisualization(DSSDocument document, SigningJob job, int pdfDpi) {
         super(job);
         this.document = document;
+        this.pdfDpi = pdfDpi;
     }
 
-    private String getBase64EncodedDocument() {
-        try (var is = document.openStream()) {
-            return new String(Base64.getEncoder().encode(is.readAllBytes()), StandardCharsets.UTF_8);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
+    private ArrayList<byte []> getPdfImages() throws IOException {
+        var pdfDocument = PDDocument.load(this.document.openStream());
+        var pdfRenderer = new PDFRenderer(pdfDocument);
+        var divs = new ArrayList<byte[]>();
+        for (int page = 0; page < pdfDocument.getNumberOfPages(); ++page) {
+            var os = new ByteArrayOutputStream();
+            var bim = pdfRenderer.renderImageWithDPI(page, pdfDpi, ImageType.RGB);
+            ImageIO.write(bim, "png", os);
+            divs.add(os.toByteArray());
         }
+
+        pdfDocument.close();
+
+        return divs;
```

Comment:
Toto ale vyrenderuje uplne vsetky strany, nie je to lazy ako to bolo predtym ze?

Reply:
Nie je to lazy loading. Neviem, ako by sme ho spravili.

Na JavaFX Image sa dá nastaviť nejaký background loading, ale musí ísť z disku. Vieme keď tak vyrenderované obrázky ukladať do nejakého temp dir a odtiaľ nastaviť loading.

Reply:
v tom html to fungovalo tak, ze som tam nahadzal placehoder divy nejakej velkosti (nacitane z prvej strany asik) a potom ked si scrollol tak to nacitalo strany +- par stran z viewportu. ten scrollpane asi ma nejake eventy ked vies, ze to je visible alebo nie.

Reply:
ale teda skusme tam 100MB pdf a ked to uvladze tak asi na to kaslime

Reply:
Sk[sil som 100MB PDF 77 strán https://cartographicperspectives.org/index.php/journal/article/view/cp43-complete-issue/pdf 

Trvalo to 13 sekúnd a pri ďalšom raze 9 sekúnd.